test/cases/compile_errors/undef_arith_returns_undef.zig

```diff
@@ -275,1773 +275,1773 @@ inline fn testFloatWithValue(comptime Float: type, x: Float) void {
+// @as(@Vector(2, u8), undefined)
+// @as(@Vector(2, u8), undefined)
 // @as(@Vector(2, u8), .{ undefined, 0 })
-// @as(@Vector(2, u8), .{ undefined, undefined })
+// @as(@Vector(2, u8), undefined)
 // @as(@Vector(2, u8), .{ undefined, 0 })
-// @as(@Vector(2, u8), .{ undefined, undefined })
-// @as(@Vector(2, u8), .{ undefined, undefined })
-// @as(@Vector(2, u8), .{ undefined, undefined })
-// @as(@Vector(2, u8), .{ undefined, undefined })
-// @as(@Vector(2, u8), .{ undefined, undefined })
+// @as(@Vector(2, u8), undefined)
+// @as(@Vector(2, u8), undefined)
+// @as(@Vector(2, u8), undefined)
+// @as(@Vector(2, u8), undefined)
+// @as(@Vector(2, u8), undefined)
 // @as(u8, undefined)
 // @as(u8, undefined)
 // @as(@Vector(2, u8), .{ 9, undefined })
 // @as(@Vector(2, u8), .{ undefined, 9 })
-// @as(@Vector(2, u8), .{ undefined, undefined })
+// @as(@Vector(2, u8), undefined)
 // @as(@Vector(2, u8), .{ 9, undefined })
 // @as(@Vector(2, u8), .{ 9, undefined })
-// @as(@Vector(2, u8), .{ undefined, undefined })
-// @as(@Vector(2, u8), .{ undefined, undefined })
+// @as(@Vector(2, u8), undefined)
+// @as(@Vector(2, u8), undefined)
 // @as(@Vector(2, u8), .{ undefined, 9 })
-// @as(@Vector(2, u8), .{ undefined, undefined })
+// @as(@Vector(2, u8), undefined)
 // @as(@Vector(2, u8), .{ undefined, 9 })
-// @as(@Vector(2, u8), .{ undefined, undefined })
-// @as(@Vector(2, u8), .{ undefined, undefined })
-// @as(@Vector(2, u8), .{ undefined, undefined })
-// @as(@Vector(2, u8), .{ undefined, undefined })
-// @as(@Vector(2, u8), .{ undefined, undefined })
+// @as(@Vector(2, u8), undefined)
+// @as(@Vector(2, u8), undefined)
+// @as(@Vector(2, u8), undefined)
+// @as(@Vector(2, u8), undefined)
+// @as(@Vector(2, u8), undefined)
 // @as(u8, undefined)
 // @as(u8, undefined)
 // @as(@Vector(2, u8), .{ 9, undefined })
 // @as(@Vector(2, u8), .{ undefined, 9 })
-// @as(@Vector(2, u8), .{ undefined, undefined })
+// @as(@Vector(2, u8), undefined)
 // @as(@Vector(2, u8), .{ 9, undefined })
 // @as(@Vector(2, u8), .{ 9, undefined })
-// @as(@Vector(2, u8), .{ undefined, undefined })
-// @as(@Vector(2, u8), .{ undefined, undefined })
+// @as(@Vector(2, u8), undefined)
+// @as(@Vector(2, u8), undefined)
 // @as(@Vector(2, u8), .{ undefined, 9 })
-// @as(@Vector(2, u8), .{ undefined, undefined })
+// @as(@Vector(2, u8), undefined)
 // @as(@Vector(2, u8), .{ undefined, 9 })
-// @as(@Vector(2, u8), .{ undefined, undefined })
-// @as(@Vector(2, u8), .{ undefined, undefined })
-// @as(@Vector(2, u8), .{ undefined, undefined })
-// @as(@Vector(2, u8), .{ undefined, undefined })
-// @as(@Vector(2, u8), .{ undefined, undefined })
+// @as(@Vector(2, u8), undefined)
+// @as(@Vector(2, u8), undefined)
+// @as(@Vector(2, u8), undefined)
+// @as(@Vector(2, u8), undefined)
+// @as(@Vector(2, u8), undefined)
 // @as(u8, undefined)
 // @as(u8, undefined)
 // @as(@Vector(2, u8), [runtime value])
 // @as(@Vector(2, u8), [runtime value])
+// @as(@Vector(2, u8), undefined)
```

Comment:
Ahhh, these changes worried me at first, but actually, this is great! Your undef canonicalization has made `.{ undefined, undefined } +% .{ runtime, runtime }` return comptime-known undef. Lovely work!